test.js

```diff
@@ -661,3 +663,180 @@ test('corrupt MKV throws', async t => {
+	const header = 'UNICORN FILE\n';
+	const uint8ArrayContent = new TextEncoder().encode(header);
+
+	const customDetectors = [unicornDetector];
+	const parser = new FileTypeParser({customDetectors});
+
+	const result = await parser.fromBuffer(uint8ArrayContent);
+	t.deepEqual(result, {ext: 'unicorn', mime: 'application/unicorn'});
+});
+
+test('fileTypeFromBuffer should keep detecting default file types when no custom detector matches', async t => {
+	// Load the content of a PNG file
+	const file = path.join(__dirname, 'fixture', 'fixture.png');
+	const uint8ArrayContent = fs.readFileSync(file);
+
+	const customDetectors = [unicornDetector];
+	const parser = new FileTypeParser({customDetectors});
+
+	const result = await parser.fromBuffer(uint8ArrayContent);
+	t.deepEqual(result, {ext: 'png', mime: 'image/png'});
+});
+
+test('fileTypeFromBuffer should allow overriding default file type detectors', async t => {
+	// Load the content of a PNG file
+	const file = path.join(__dirname, 'fixture', 'fixture.png');
+	const uint8ArrayContent = fs.readFileSync(file);
+
+	const customDetectors = [mockPngDetector];
+	const parser = new FileTypeParser({customDetectors});
+
+	const result = await parser.fromBuffer(uint8ArrayContent);
+	t.deepEqual(result, {ext: 'mockPng', mime: 'image/mockPng'});
+});
+
+class CustomReadableStream extends stream.Readable {
+	_read(_size) {
+		this.push('UNICORN');
+	}
+}
+test('fileTypeFromStream should detect custom file type "unicorn" using custom detectors', async t => {
+	// Convert the "unicorn" file content to a readable stream
+	const readableStream = new CustomReadableStream();
+
+	const customDetectors = [unicornDetector];
+	const parser = new FileTypeParser({customDetectors});
+
+	const result = await parser.fromStream(readableStream);
+	t.deepEqual(result, {ext: 'unicorn', mime: 'application/unicorn'});
+});
+
+test('fileTypeFromStream should keep detecting default file types when no custom detector matches', async t => {
+	// Load the content of a PNG file
+	const file = path.join(__dirname, 'fixture', 'fixture.png');
+	const readableStream = fs.createReadStream(file);
+
+	const customDetectors = [unicornDetector];
+	const parser = new FileTypeParser({customDetectors});
+
+	const result = await parser.fromStream(readableStream);
+	t.deepEqual(result, {ext: 'png', mime: 'image/png'});
+});
+
+test('fileTypeFromStream should allow overriding default file type detectors', async t => {
+	// Load the content of a PNG file
+	const file = path.join(__dirname, 'fixture', 'fixture.png');
+	const readableStream = fs.createReadStream(file);
+
+	const customDetectors = [mockPngDetector];
+	const parser = new FileTypeParser({customDetectors});
+
+	const result = await parser.fromStream(readableStream);
+	t.deepEqual(result, {ext: 'mockPng', mime: 'image/mockPng'});
+});
+
+test('fileTypeFromFile should detect custom file type "unicorn" using custom detectors', async t => {
+	// Load unicorn file content
```

Comment:
This line is not "loading" anything, and I don't think the comment adds anything. Same with the other tests.